tests/unit/src/test/scala/tests/debug/StackFrameDapSuite.scala

```diff
@@ -0,0 +1,159 @@
+    source = """|a/src/main/scala/Main.scala
+                |object Main {
+                |  def main(args: Array[String]): Unit = {
+                |>>  println()
+                |  }
+                |}
+                |""".stripMargin,
+    expectedFrames = List(
+      Variables(
+        Scope.local(Variable("this: Main$"), Variable("args: String[]"))
+      )
+    )
+  )
+
+  assertStackFrame("primitives")(
+    source = """|a/src/main/scala/Main.scala
+                |object Main {
+                |  def main(args: Array[String]): Unit = {
+                |    foo()
+                |  }
+                |
+                |  def foo(): Unit = {
+                |    val aByte = 1.toByte
+                |    val aShort = 1.toShort
+                |    val anInt  = 1
+                |    val aLong  = 1L
+                |    val aFloat = 1.0f
+                |    val aDouble = 1.0
+                |    val bool = true
+                |    val aChar = 'a'
+                |>>  println()
+                |  }
+                |}
+                |""".stripMargin,
+    expectedFrames = List(
+      Variables(
+        Scope.local(
+          Variable("aByte: byte = 1"),
+          Variable("aShort: short = 1"),
+          Variable("anInt: int = 1"),
+          Variable("aLong: long = 1"),
+          Variable("aFloat: float = 1.000000"),
+          Variable("aDouble: double = 1.000000"),
+          Variable("bool: boolean = true"),
+          Variable("aChar: char = a"),
+          Variable("this: Main$")
+        )
+      )
+    )
+  )
+
+  assertStackFrame("for-comprehension")(
+    source = """|a/src/main/scala/Main.scala
+                |object Main {
+                |  def main(args: Array[String]): Unit = {
+                |    for {
+                |      x <- List(1)
+                |>>    z = x + 2
+                |    } println(z)
+                |  }
+                |}
+                |""".stripMargin,
+    expectedFrames = List(
+      Variables( // before calculating `z`
+        Scope.local(Variable("x: int = 1"))
+      ),
+      Variables( // after calculating `z`
+        Scope.local(Variable("x: int = 1"), Variable("z: int = 3"))
+      ),
+      Variables(Scope.local(Variable("x$1: Tuple2$mcII$sp")))
+    )
+  )
+
+  assertStackFrame("overridden-toString")(
+    source = """|a/src/main/scala/Main.scala
+                |object Main {
+                |  def main(args: Array[String]): Unit = {
+                |    val foo = new Foo
+                |>>  println()
+                |  }
+                |}
+                |
+                |class Foo {
+                |  override def toString = "foo"
+                |}
+                |""".stripMargin,
+    expectedFrames = List(
+      Variables(
+        Scope.local(
+          Variable("this: Main$"),
+          Variable("args: String[]"),
+          Variable("foo: Foo")
+        )
+      )
+    )
+  )
+
+  def assertStackFrame(
```

Comment:
Can we add a test case with symbolic identifiers? I noticed that `::` was displayed as `$colon$colon` in the variable view

<img width="389" alt="Screenshot 2019-12-16 at 11 03 07" src="https://user-images.githubusercontent.com/1408093/70903058-f836a200-1ff5-11ea-9732-cb89f65bb37c.png">

No need to change the current behavior, just document it with a test case.

Reply:
Did we add a test case with symbolic identifier?

Reply:
Yup: https://github.com/scalameta/metals/pull/1163/files#diff-e39244aad5f9193c28b3779716ed886bR124